runtime/runtime/src/actions.rs

```diff
@@ -1329,6 +1463,58 @@ mod tests {
         )
     }
 
+    #[test]
+    fn test_delete_account_removes_gas_keys() {
+        let account_id: AccountId = "alice".parse().unwrap();
+        let public_keys: Vec<PublicKey> =
+            (0..3).map(|_| SecretKey::from_random(KeyType::ED25519).public_key()).collect();
+        let gas_key = GasKey {
+            balance: Balance::from_yoctonear(10),
+            num_nonces: 2,
+            permission: AccessKeyPermission::FullAccess,
+        };
+        // Setup account with gas key
+        let tries = TestTriesBuilder::new().build();
+        let mut state_update = setup_account_with_tries(
+            &tries,
+            &account_id,
+            &public_keys[0],
+            &AccessKey { nonce: 0, permission: AccessKeyPermission::FullAccess },
+        );
+        for public_key in &public_keys {
+            set_gas_key(&mut state_update, account_id.clone(), public_key.clone(), &gas_key);
+            for i in 0..gas_key.num_nonces {
+                set_gas_key_nonce(&mut state_update, account_id.clone(), public_key.clone(), i, 0);
+            }
+        }
+        state_update.commit(StateChangeCause::InitialState);
+        let trie_changes = state_update.finalize().unwrap().trie_changes;
+        let mut store_update = tries.store_update();
+        let root = tries.apply_all(&trie_changes, ShardUId::single_shard(), &mut store_update);
+        store_update.commit().unwrap();
+
+        // Delete account
+        let mut state_update = tries.new_trie_update(ShardUId::single_shard(), root);
+        let action_result =
+            test_delete_large_account(&account_id, &CryptoHash::default(), 100, &mut state_update);
+        assert!(action_result.result.is_ok());
+
+        // Verify gas key and nonces are removed
```

Comment:
This comment seems misplaces, IIUC below we only finalize the state update and only after `        store_update.commit().unwrap();` we check that there are no more gas keys.

Reply:
As an optional: I would get rid of these comments altogether, but up to you.

Reply:
I moved the comment right before the verification.